src/test/java/com/vinyl/LoggedInBaseIntegration.java

```diff
@@ -0,0 +1,55 @@
+package com.vinyl;
+
+import com.vinyl.helper.TokenHeaderHelper;
+import com.vinyl.model.Product;
+import com.vinyl.repository.TokenRepository;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+
+import java.time.LocalDate;
+
+public abstract class LoggedInBaseIntegration extends BaseIntegration {
+
+    Product product;
+
+    @Autowired
+    TokenHeaderHelper tokenHeaderHelper;
+
+    @Autowired
+    TokenRepository tokenRepository;
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        product = defaultEntitiesHelper.createProduct();
```

Comment:
actually move the protected Token token; field from BaseIntegration here, and do the setUp for it here as well

Reply:
actually I am gonna need the token field in BaseIntegeration for the AddProductToCart test which does not extend LoggedInBaseIntegration